### PR TITLE
[core] Add API to set camera when initializing map

### PR DESCRIPTION
**Pull request by jmalanen**
Allows initializing map with given camera options.

### CODE REVIEW

File: include/mbgl/map/map_options.hpp [hunk 1]
```diff
@@ -134,6 +134,9 @@ class MapOptions final {
      */
     float pixelRatio() const;
 
+    MapOptions& withCameraCenter(LatLng latLng);
```
Comment (tmpsantos) [inline include/mbgl/map/map_options.hpp]: what about doing `withCamera`?

Comment (jmalanen) [reply include/mbgl/map/map_options.hpp]: Changed

File: src/mbgl/style/style_impl.cpp [hunk 2]
```diff
@@ -36,6 +36,16 @@ Style::Impl::Impl(FileSource& fileSource_, float pixelRatio)
     light->setObserver(this);
 }
 
+Style::Impl::Impl(FileSource& fileSource_, float pixelRatio, CameraOptions camera)
+    : Impl(fileSource_, pixelRatio) {
```
Comment (alexshalamov) [inline src/mbgl/style/style_impl.cpp]: why not to std::move camera into defaultCamera?

File: src/mbgl/style/style.cpp [hunk 3]
```diff
@@ -12,6 +12,10 @@ Style::Style(FileSource& fileSource, float pixelRatio)
     : impl(std::make_unique<Impl>(fileSource, pixelRatio)) {
 }
 
+Style::Style(FileSource& fileSource, float pixelRatio, CameraOptions camera)
+    : impl(std::make_unique<Impl>(fileSource, pixelRatio, camera)) {
```
Comment (alexshalamov) [inline src/mbgl/style/style.cpp]: std::move(camera)

File: src/mbgl/map/map_options.cpp [hunk 4]
```diff
@@ -81,4 +82,13 @@ float MapOptions::pixelRatio() const {
     return impl_->pixelRatio;
 }
 
+MapOptions& MapOptions::withCamera(CameraOptions camera_) {
+    impl_->camera = camera_;
```
Comment (alexshalamov) [inline src/mbgl/map/map_options.cpp]: impl_->camera = std::move(camera_)

File: src/mbgl/style/style_impl.cpp (hunk 2 above)
Comment (jmalanen) [reply src/mbgl/style/style_impl.cpp]: Done

File: src/mbgl/style/style.cpp (hunk 3 above)
Comment (jmalanen) [reply src/mbgl/style/style.cpp]: Done

File: src/mbgl/map/map_options.cpp (hunk 4 above)
Comment (jmalanen) [reply src/mbgl/map/map_options.cpp]: Done

File: include/mbgl/map/map_options.hpp [hunk 5]
```diff
@@ -134,6 +135,21 @@ class MapOptions final {
...
+     * @return reference to MapOptions for chaining options together.
+     */
+    MapOptions& withCamera(CameraOptions camera_);
+
+    /**
+     * @brief Gets the previously set camera options.
+     *
+     * @return Camera options.
+     */
+    CameraOptions camera() const;
```
Comment (pozdnyakov) [inline include/mbgl/map/map_options.hpp]: `const CameraOptions&` ?

Comment (jmalanen) [reply include/mbgl/map/map_options.hpp]: Done

**Comment by tmpsantos** [comment]
Missing unit tests and we are good for merging.

**Comment by astojilj** [comment]
Good to add some description or track to another issue. Where does the requirement come from?

**Comment by chloekraw** [comment]
> Good to add some description or track to another issue. Where does the requirement come from?

@astojilj  👉  https://github.com/mapbox/mapbox-gl-native-team/issues/89

**Comment by jmalanen** [comment]
The necessity of this needs to be re-evaluated. It should be possible to achieve the same result with the existing APIs by using `Map::jumpTo()` before `Map::setStyle()`. Therefore the new API would be redundant.

**Pull request closed without merging**